agent_pool.go

```diff
@@ -91,6 +94,12 @@ type AgentPoolCreateOptions struct {
 
 	// Required: A name to identify the agent pool.
 	Name *string `jsonapi:"attr,name"`
+
+	// True if the agent pool is organization scoped, false otherwise.
+	OrganizationScoped *bool `jsonapi:"attr,organization-scoped,omitempty"`
+
+	// List of workspaces that are associated with an agent pool.
```

Comment:
I don't see the `omitempty` tag some I am guessing that  `AllowedWorkspaces` is required. Can you add to the comment that this is required?: `// Required: List of workspaces that are associated with an agent pool.`